### DISCUSSION TITLE
Need a hand for Astroid 3.0.3 Testing

### DISCUSSION THREAD

**Discussion by sonvnn** [templaza/astroid-framework#425] [Announcements]
Hi All, @Chacapamac , @Joomlaplates ,

I am working on Astroid 3.0.3. You can review this version below:

- Migrate all function from Joomla 4 to Joomla 6.
- Drop Joomla 3
- Optimize Loading assets 
- Improve UI/UX for Layout Builder
- Improve UI/UX for Custom Code Editor

We need you help to test this with Joomla 5 latest. 
### How to test

1. Clone your site to a test server (Don't do it on live site)
2. Download/Install Astroid 3 dev https://github.com/templaza/astroid-framework/archive/refs/heads/v3.zip
3. Download/Install One template https://github.com/templaza/astroid-framework/archive/refs/heads/one-template.zip
4. Go to Administrator -> System -> Plugins -> Find "Behaviour - Backward Compatibility" Plugin
5. Disable Behaviour - Backward Compatibility
6. Test all function from backend to frontend
7. Please tell me any issue you got.

Thank you all! You are great! We will make Astroid great again, together.

Best Regards,
Sonny

**Comment by darekmer** [comment]
Hi, clean Joomla 5.0 AF 3.0.3 PHP 8.2, Disabled Behaviour - Backward Compatibility
jQuery.Deferred exception: $(...).find(...).astroidMobileMenu is not a function initOffcanvasMenu@https://pi.local/j5/media/astroid/assets/vendor/astroid/js/script.js?v=3.0.2:24:47
docReady@https://pi.local/j5/media/astroid/assets/vendor/astroid/js/script.js?v=3.0.2:299:7
e@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:27028
Deferred/then/l/</t<@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:27330
setTimeout handler*Deferred/then/l/<@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:27594
c@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:25304
fireWith@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:26053
fire@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:26089
c@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:25304
fireWith@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:26053
ready@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:29112
P@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:28884
EventListener.handleEvent*@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:29264
@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:220
@https://pi.local/j5/media/vendor/jquery/js/jquery.min.js?3.7.1:2:225
 undefined jquery.min.js:2:28648

other:
Uncaught TypeError: $(...).find(...).astroidMobileMenu is not a function
    initOffcanvasMenu https://pi.local/j5/media/astroid/assets/vendor/astroid/js/script.js?v=3.0.2:24
    docReady https://pi.local/j5/media/astroid/assets/vendor/astroid/js/script.js?v=3.0.2:299
    jQuery 13
script.js:24:47

**Reply by sonvnn** [reply]
Please help me re-download v3 and test it again. Thank you.

**Reply by darekmer** [reply]
after installing, I opened the home page and only Preloader works and works and nothing opens, clear cache did not help
![wisi](https://github.com/templaza/astroid-framework/assets/44267306/83848b93-7051-4ccf-8f20-122f70fd7695)

**Reply by darekmer** [reply]
clean install 5.0 >> 2 minutes ago, only Preloader
![clean](https://github.com/templaza/astroid-framework/assets/44267306/4bdfbe99-3a05-4677-978f-fa762fc50fed)

**Reply by sonvnn** [reply]
@Chacapamac @Joomlaplates please help me for testing 3.0.3. I need more testing result.

Thank you all!

**Reply by sonvnn** [reply]
@darekmer Sorry I forgot move js files. Just update v3-branch. Please help me download and check it again. Thank you.

**Reply by darekmer** [reply]
Hi, it's ok now

![clean2](https://github.com/templaza/astroid-framework/assets/44267306/ea493fa0-ba7d-445a-a2b9-a58bf99f0018)

**Reply by sonvnn** [reply]
@darekmer let's test everything and tell me if you find any issue. 

Thank you :)

**Reply by darekmer** [reply]
no errors, but bootstrap tabs and accordion don't work, sometimes they work, sometimes they don't

**Reply by sonvnn** [reply]
which page do you get it?

**Reply by darekmer** [reply]
ip address on mail, modal not working
tell me when to turn it off
links:
https://x.x.x.x/j5a/blog/prawo/ustawa-o-rod
https://x.x.x.x/j5a/j5a/blog/prawo/przewodnik-po-ustawie
home page
hi, I wasn't there. I turned on access

**Reply by sonvnn** [reply]
May you give me a screenshot or video screen record? Thanks

**Comment by darekmer** [comment]
"Sign in with a passkey" in cassiopeia working, one template not working
Uncaught TypeError: e.renderMessages is not a function
    plgSystemWebauthnLogin https://pi.local/j5a/media/plg_system_webauthn/js/login.min.js?6c425f:1
    <anonymous> https://pi.local/j5a/media/plg_system_webauthn/js/login.min.js?6c425f:1
    EventListener.handleEvent* https://pi.local/j5a/media/plg_system_webauthn/js/login.min.js?6c425f:1
    <anonymous> https://pi.local/j5a/media/plg_system_webauthn/js/login.min.js?6c425f:1
    <anonymous> https://pi.local/j5a/media/plg_system_webauthn/js/login.min.js?6c425f:1
login.min.js:1:934

**Reply by sonvnn** [reply]
Please help me download v3-branch and test again. Thanks

**Comment by joomlaplates-de** [comment]
Hi Sonny, custom fonts are not working.
![1_2023-10-27](https://github.com/templaza/astroid-framework/assets/71080509/33062b4c-f16f-44eb-b666-5dfaf1ee086d)

**Reply by sonvnn** [reply]
Thank you! I fixed it in https://github.com/templaza/astroid-framework/commit/410f19850006e6c10ec19d51f58f625aa96b61f9

**Comment by sonvnn** [comment]
Thank you all! After research everything I decide Astroid 3.0.3 will drop Joomla 3 Support. All website sites are running on Joomla 3 can use Astroid 3.0.2 without any issue, but they can not update to Astroid 3.0.3

**Reply by darekmer** [reply]
then for joomla 3.x create the last version 2.9.9 and notify that this is the last version for joomla 3.x.

**Reply by Chacapamac** [reply]
It is time to move on.

**Comment by deltapapa01** [comment]
Hi @sonvnn ,
found an error. When you select an image in preloader, FE crashed:

![image](https://github.com/templaza/astroid-framework/assets/19833064/8a9045a5-121b-48ae-9c77-07137749a10f)

Class "JURI" not found in /html/cms/templates/astroid_template_zero/error.php:75

**Reply by sonvnn** [reply]
Please test on Astroid_template_one first. I did not update zero yet. 

Thank you.

**Reply by deltapapa01** [reply]
Okay, I will. Unfortunately, I created all my pages with the One. Where is the difference actually? Wouldn't it be easier and clearer if there is only 1 basic template?

**Reply by deltapapa01** [reply]
Hi @sonvnn,
i install template one, and get the follow error direct after install:

![image](https://github.com/templaza/astroid-framework/assets/19833064/b170da0d-18d8-4606-8e53-56fec72474db)

I use this version: https://github.com/templaza/astroid-framework/archive/refs/heads/one-template.zip

**Reply by deltapapa01** [reply]
And in template manager from joomla this:

![image](https://github.com/templaza/astroid-framework/assets/19833064/26e02812-f9a5-4904-92d0-b0e77ff7f34d)

**Reply by deltapapa01** [reply]
Sorry, and with b7c plugin i get the follow error, when i open astroid:
![image](https://github.com/templaza/astroid-framework/assets/19833064/b9b12eb7-3434-4d74-b17d-ad811177627b)

**Reply by sonvnn** [reply]
I fixed this issue please help me download v3-retest again.

**Reply by sonvnn** [reply]
Please help me download One Template here and retest https://github.com/templaza/astroid-framework/archive/refs/heads/one-template.zip

Thanks

**Comment by deltapapa01** [comment]
Is this px or ms?

![image](https://github.com/templaza/astroid-framework/assets/19833064/480ab2f5-35f2-4596-b0cc-c63b41d62c1c)

**Comment by deltapapa01** [comment]
One more thing:
If you load present style, the scroll speed is gone:
![image](https://github.com/templaza/astroid-framework/assets/19833064/b99e7317-4ba5-426c-8889-d49cde8c6bae)

If you disable and enabled Smooth Scroll speed, it is showing:
![image](https://github.com/templaza/astroid-framework/assets/19833064/5eca1256-25a9-428a-8db3-6735ba11c5af)

**Reply by sonvnn** [reply]
1. Cause is preset is old data. We will update it in the next version.
2. I fixed this issue in https://github.com/templaza/astroid-framework/commit/61762b137a3d87b115168b16fabd8de347aa3985

**Reply by deltapapa01** [reply]
@sonvnn 
i can't see any difference at all with the smooth scroll, no matter what value i set for ms. How can I test this?
The labeling now fits

**Comment by deltapapa01** [comment]
So, when enabled Open Graph, FE is broken too:
![image](https://github.com/templaza/astroid-framework/assets/19833064/8c1e4767-3915-4718-9a2b-f51a400c13b2)

![image](https://github.com/templaza/astroid-framework/assets/19833064/fb39da22-29bc-4b18-bde6-18c27188e57c)

**Reply by deltapapa01** [reply]
@sonvnn 
test it with One, no error! Hope you will update Zero too....

**Comment by deltapapa01** [comment]
I don't no why, but when i set SASS Overrides, nothing change to my site:

![image](https://github.com/templaza/astroid-framework/assets/19833064/f19e3586-58b0-45bc-8a67-958feb491c34)

I delete all caches, but it is still the primary color from color menu Musst i do somthing to override colors?

**Reply by sonvnn** [reply]
Bootstrap is updated to v5.3. Please check if these variables is available in new version.

**Reply by deltapapa01** [reply]
@sonvnn 
yepp, i test it with one and zero, it works fine. I am not sure, maybe it was my fault.

Small other question (or feature request)
Is it possible, to set colors to bootstrap variables?
For example, it was amazing when i can use here primary, secondary or any other color classes from bootstrap, so i can chance all the colors on my site with only one setting:

![image](https://github.com/templaza/astroid-framework/assets/19833064/da34c049-0fe9-4f5b-903b-de3fd7878265)

**Comment by deltapapa01** [comment]
Okay, and i get one more error, but i don't no what exactly i did :-)
![image](https://github.com/templaza/astroid-framework/assets/19833064/d0edb3d8-d687-41d3-bc04-33e70558b49d)

So, i will pause my test now, it's time to go swimming. Today we have in german a holiday :-)

**Reply by sonvnn** [reply]
I fixed this issue. Please download v3 and retest it. Thanks.

**Reply by deltapapa01** [reply]
@sonvnn 
Test since one hour with zero and One, this Issue has not occurred again until now!

**Comment by deltapapa01** [comment]
Hi Sonny,

I have made the following discovery, which could possibly be a security problem. When I turn on the Author Info in the template options, I am automatically logged into the front end. If I turn the option off again or turn it on again, I stay logged into the frontend all the time.
I think this is a behavior which is not wanted.

Take a look here: https://1drv.ms/v/s!Ar1_3t0lo0q1ifA3j40qxDNETbk9Gw?e=kygYS6

**Reply by darekmer** [reply]
same for me
when I click on the title of the article, the full article will open and automatically log in the admin

**Reply by sonvnn** [reply]
@darekmer @deltapapa01 Thank you for report security issue. I fixed this issue in https://github.com/templaza/astroid-framework/commit/4d307029b036b55d5b73df2afc2e712a4e8ce4f8

Please help me download v3-branch and retest again.

Thank you

**Reply by deltapapa01** [reply]
@sonvnn 
Hi, i test it in J4.4.0 and J5.0, this issue is gone. Perfect!

**Comment by darekmer** [comment]
ip address on mail, modal not working
tell me when to turn it off
links:
https://x.x.x.x/j5a/blog/prawo/ustawa-o-rod
https://x.x.x.x/j5a/j5a/blog/prawo/przewodnik-po-ustawie
home page tabs, on mobile accordion
I turned off access

**Reply by sonvnn** [reply]
I can not access to your server via ip.

**Reply by darekmer** [reply]
I opened access on the router, write to me when you're done

**Reply by sonvnn** [reply]
Thank you! I will update you soon.

**Reply by darekmer** [reply]
https://drive.google.com/file/d/18tg7G-qfjY9ytb0TORX00vK4LmLZi6rl/view?usp=sharing

**Reply by sonvnn** [reply]
I know the issue because you create a custom html in article. In default, Astroid does not load all Bootstrap JS. It loads bootstrap js depend of its purpose. Ex: Bootstrap.carousel for Article slider, Bootstrap.offcanvas for Offcanvas

I will find a way to help admin have ability can choose which is Bootstrap library will be load in default.

You can turn off access.

Best regards,
Sonny

**Reply by darekmer** [reply]
Now I just noticed that the 'bootstrap bundle was not loaded' ;(

**Reply by sonvnn** [reply]
Hi

Now you can bootstrap js here 

![Screen Shot 2023-11-01 at 16 41 20](https://github.com/templaza/astroid-framework/assets/3204103/2b763df0-29d0-4afd-af80-d3250a49d031)

Let's choose the libraries you want to load in template default. Please help me re-download v3-branch for testing.

Thanks,

**Reply by darekmer** [reply]
Bootstrap Tabs, Accordion, Modal works.
Bravo you 
![bravo](https://github.com/templaza/astroid-framework/assets/44267306/8e97d85f-83e8-433a-84e6-44d453273e33)

**Comment by Chacapamac** [comment]
@sonvnn 
What version of Joomla I need to install to test Astroid 3.0.3?

**Reply by joomlaplates-de** [reply]
Joomla 4 or Joomla 5

**Reply by sonvnn** [reply]
You can check it on Joomla 4 and Joomla 5.

Note: in Joomla 5 please disable B/C plugin. And just test on One template only.

Please hurry we need more testing for this version.

**Comment by joomlaplates-de** [comment]
Joomla 4.4
Template One 3.0.3
**Astroid Setting > Header > Mobile Menu = OFF**
I get a jQuery error and the "to top" scroll isnt available.
![jqzury](https://github.com/templaza/astroid-framework/assets/71080509/41eddc1f-8940-4f84-b247-068ab7edcadf)

**Reply by sonvnn** [reply]
I fixed this issue in https://github.com/templaza/astroid-framework/commit/2ca62dcd9b1ebca6f90e4fe3419f8a6357e2472f

**Comment by darekmer** [comment]
Updating '404 error' will also be useful
This is how it looks to me in Joomla 3.x:
![j3-404](https://github.com/templaza/astroid-framework/assets/44267306/14c68109-a17f-419c-a8fb-85b11ef527e7)
And this is what it looks like in Joomla 5:
![j5-404](https://github.com/templaza/astroid-framework/assets/44267306/c0738349-4e76-4305-b380-f7166acbafcc)

**Reply by darekmer** [reply]
AF settings do not work: Background Type, Background Color, Background Image.

**Reply by sonvnn** [reply]
It works perfect on my local. May you give me your link again?

**Comment by joomlaplates-de** [comment]
Joomla 4.4
Template ONE
....
Selecting images is not possible
![404](https://github.com/templaza/astroid-framework/assets/71080509/974b9438-a905-47e9-8516-722a36f07eb9)

**Reply by sonvnn** [reply]
I fixed this issue in https://github.com/templaza/astroid-framework/commit/add93aab4a0bd305d6249f3d2fd312fd81e1819a#diff-bfa32a49fbe93a95df7e5b078ba3c1cc1ae74d274adc72aed5a84a954f5efd37

Please help me download v3 and re-check again.

Thanks,
Sonny

**Reply by joomlaplates-de** [reply]
Hi Sonny,
1.) Its still not possible to select an image, see picture attached
2.) I used your donwload link. But please if you update the Astroid DEV, change the version number to 
- 3.0.3.1
- or 3.0.3a or 3.0.3b
- Otherwise nobody knows what is the newest DEV version and we get all confused.
- Regards Peter. Great JOB!
- 
![Screenshot_1](https://github.com/templaza/astroid-framework/assets/71080509/d9e6c36b-4172-419f-8b17-2ab92dc706ae)

**Reply by sonvnn** [reply]
In your attached image is video media not image. Please send me a link I will check it on your server.

Thanks,

**Reply by joomlaplates-de** [reply]
Hi Sonny, doesnt matter if its a video or image.
Media folder isnt shown.
Ive send ou the link

**Reply by sonvnn** [reply]
@Joomlaplates I check my email sonlv@template.com. But I did not received your email yet.

**Reply by sonvnn** [reply]
@Joomlaplates It is missing function in Joomla 4.4. I fixed this issue in https://github.com/templaza/astroid-framework/commit/93db37c3b2c20fcbed7c6a43702b5c71f933aa23 I updated code on your test site. Please help me check it.

**Reply by joomlaplates-de** [reply]
Yes its working now! Was a bug in Astroid 3.0.3?

**Reply by sonvnn** [reply]
It's a bug of Joomla 4.4. But I have fixed it by an alterlative function, it may deprecated in Joomla 6. I will create a pull request for Joomla project tomorrow.

**Comment by Chacapamac** [comment]
**◉ Ok I’m on it**
For now:
• Joomla 4.4.0 _(Before upgrading to 5)_
• Astroid 3.0.2 
• PHP 8.1.0
• Plugin “**Behaviour - Backward Compatibility**” OFF

**First remark:**
On Joomla **Extensions Pre-Update Check** I get this warning?
![Screen Shot 2023-11-05 at 9 48 16 AM](https://github.com/templaza/astroid-framework/assets/1907276/308e485b-5ffb-4db0-bff0-f324a725cd26)

————————————————————————————————————————
◉ UPDATE

**1. I just update to the Dev version show at the start of the post**
Download/Install Astroid 3 dev https://github.com/templaza/astroid-framework/archive/refs/heads/v3.zip
Download/Install One template https://github.com/templaza/astroid-framework/archive/refs/heads/one-template.zip

— and I did have the same **Extensions Pre-Update Check** message before updating to Joomla 5

**2. I update to Joomla 5 without problems**

READY TO TEST

**Reply by sonvnn** [reply]
I updated Astroid Plugin follow latest style of Joomla 5 https://github.com/templaza/astroid-framework/commit/097f840d205f12f8f9aec4081965da1f48bc7791

I am not sure if it works.

**Comment by sonvnn** [comment]
Hello, Please check everything one more time. If no issue discovered in next 24h. I will release Astroid 3.0.3

Thanks & Best Regards,
Sonny

**Comment by rinka88** [comment]
Is it possible when compiling there were not many css files, there was only one. and the extra ones were deleted by itself?

**Reply by sonvnn** [reply]
You can use compress css feature to merge all css to one.

**Comment by Chacapamac** [comment]
**So far no problem or error with**
• Joomla 4.4.0 (Before upgrading to 5)
• Astroid 3.0.2
• PHP 8.1.0
• Plugin “Behaviour - Backward Compatibility” OFF

**The only errors in the console** is when I go in a menu with articles (like blog or single article) on my demo).

**• No error on the “Home” page** —> http://joom4-astro.graf-test.ca/

• Errors on the “My Blog” or any other pages (Not “Home”) —> http://joom4-astro.graf-test.ca/blog

**Not sure if related to Astroid????**

![Screen Shot 2023-11-07 at 12 33 18 PM](https://github.com/templaza/astroid-framework/assets/1907276/b532492f-0b10-4994-8a71-6eaa9aa73e38)

**Reply by sonvnn** [reply]
Because google font updated your font style. Let's go to template options and re-select font in typography tab. Save template - done!

**Reply by Chacapamac** [reply]
Yep! No More errors
Thanks @sonvnn 
Have a great day!

**Reply by joomlaplates-de** [reply]
Yes no error anymore. Great job. Impressive
But I see a problem with automatic update and Joomla 3 users!!
Maybe add a message, that the Astroid 3.0.3 cant use for Joomla 3. Otherwise J3 users update and get an error.
Regards Peter

**Comment by deltapapa01** [comment]
Hi @sonvnn ,
looks very good, i test a lot with no more errors. Only PreLoader is flashing the site on loading, this was happend with all AF Versions. And i hope, you will update Zero soon, because i use on all my sites Zero 😁. Or is there a way to  export from Zero and import the settings to One?

Thanky a lot for your great work,
WBR,
deltapapa

**Reply by joomlaplates-de** [reply]
Hello, you can import your settings into Template "ONE" and it should look the same.
Zero Template and One template are the same, just different Styles ( json files)

**Reply by sonvnn** [reply]
I am working on Two template. I will update One and Zero after that. I am working to rebuild everything of Astroid. It's an envolution. At the moment, you can still use B/C to work with Zero. I will release Two in the end of this month.

**Reply by deltapapa01** [reply]
Cool, but did I understand you correctly? You want to offer a third template? It's a bit complicated for the simple user to recognize the differences between the templates.
Isn't it easier if you offer the framework and one template? If they only differ in color anyway. Or in the docs there should be an explanation of which is the recommended, default template, which makes the decision easier for the user.
For example, I installed both templates when testing at the beginning and found no differences except for the colors.
And then I thought that Zero was the mother template, so I used it everywhere.

**Reply by sonvnn** [reply]
I am working for new free Astroid template named "Two". One and Zero will be soon updated according new template release. They will run fast as Flash of Marvel. Let' wait for update 😄

**Comment by Chacapamac** [comment]
> They will run fast as Flash of Marvel
ASTROID is NUMBER 1 (and 2!) 🏆🏁

**Comment by rinka88** [comment]
> Я работаю над шаблоном Two. После этого я обновлю One и Zero. Я работаю над восстановлением всего Астроида. Это инволюция. На данный момент вы все еще можете использовать B/C для работы с Zero. Я выпущу Two в конце этого месяца.
today is the last day of the month, waiting for more announcement


today is the last day of the month, waiting for more announcement😊
